### DISCUSSION TITLE
How Can I execute computed value only once in multiple[views]

### DISCUSSION THREAD

**Discussion by songzhenqi** [mobxjs/mobx-state-tree#2102] [Q&A] [answered]
![image](https://github.com/mobxjs/mobx-state-tree/assets/9973072/2a132437-da76-4b27-809f-baaa824dfb88)

in this case, I expect that  **valuePlusTime**  execute  only once but actually not.

am i wrong in somewhere?

how can i fix that

code :
````javascript
import { types } from "mobx-state-tree";


const Model = types
    .model({
        id: types.identifier,
        value: types.number,
        time: types.number
    })
    .views(self => ({
        get valuePlusTime() {
            console.log('update');
            return self.value + self.time;
        }
    }))
    .views(self => ({
        get result() {
            return self.valuePlusTime + self.time;
        }
    }));

const m = Model.create({
    id: '1',
    value: 1,
    time: 2
});

console.log(m.result);
console.log(m.result);
````

**Comment by songzhenqi** [comment]
should wrap the console call in Mobx like

````javascript
 when(() => true, () => {
        console.log(m.result);
        console.log(m.result);
    })
````

**Comment by dbstraight** [comment]
Can someone explain this in more detail? I'm also dealing with a variant of this problem.

**Reply by coolsoftwaretyler** [reply, marked as answer]
Hey @dbstraight - just linking to my [in depth answer in your newer discussion post](https://github.com/mobxjs/mobx-state-tree/discussions/2161#discussioncomment-8708915) in case other people end up here.
